### DISCUSSION TITLE
Support for type information in lint rules?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [rome/tools#3965] [Suggestions]
Many of the rules from [typescript-eslint](https://typescript-eslint.io) use TypeScript type checking APIs to make inferences on code that wouldn't have been possible from just syntax alone [[Linting with Type Information docs](https://typescript-eslint.io/linting/typed-linting)). Rules that use type information are generally slower than pure-AST-based ones for having to query the type checker, but can be quite powerful as a result. For example, [`@typescript-eslint/await-thenable`](https://typescript-eslint.io/rules/await-thenable) detects `await`s on values that are not Thenable.

Are there any plans to try to make that classification of rules available in Rome? Say, calling Rust-to-JS to ask the TypeScript type checker for node information?

**Comment by leops** [comment]
Yes, this is something we plan to explore in the following months as we expand our analysis infrastructure.
